### DISCUSSION TITLE
Naming standard for all abilities

### DISCUSSION THREAD

**Discussion by soloxcx** [genshinsim/gcsim#995] [Ideas / Features Request]
I propose that we rename every AttackInfo's `Abil` to a standardized format, but I'm not sure of what it should be. Right now, the naming convention is kind of all of the place.

My first thought is to use something like `Exact Ability Name (Attachment 1) [Attachment 2]`. Attachments are either additional information that's necessary to distinguish between different parts of the same ability, or it's constellation/ascension.

e.g. `Fischl C6` would become `Joint Attack (C6)`.

The exact ability name comes from the Genshin Wiki and should be the specific descriptor for the attack rather than the talent name, which can sometimes be found in the Advanced Properties part of Gameplay Notes. If there is no obvious or reasonable descriptor, then use the talent name. Ganyu's skill is called Trail of the Qilin, but the damage instance is an Ice Lotus with the attachments (Activation) and (Explosion).

Do we think the constellations/ascension attachments are actually necessary, as opposed to just learning the name?

Should we use brackets for the rare case where multiple attachments are needed, and the constellation or ascension should be last?

e.g. `C2 Mini-Frostflake Seki no To (Cutting)` -> `Additional Frostflake Seki no To (Cutting) [C2]`?

**Comment by kylebartz** [comment]
As we improve the stats exposed on the UI, our naming strategy here will directly impact how this information is exposed to users. Definitely important to think about how we want this information to also be used.

One thought I have is if we can, we should have a consistent naming strategy between what we show and another source such as project amber, kqm, wiki, etc. This would allow us to cross link between sim stats of an ability and what the official damage and scaling numbers are. IE: I see that "Kamisato Art: Soumetsu" is a big damage contributor to a sim and I can click it and see its damage scaling, description, etc.
